src/tag.ts

```diff
@@ -108,12 +108,31 @@ export class Tag implements ToCBOR, Decodeable {
    * @param tag Tag number.
    * @returns Old decoder, if there was one.
    */
-  public static clearDecoder(tag: number): TagDecoder | undefined {
+  public static clearDecoder(tag: TagNumber): TagDecoder | undefined {
     const old = this.#tags.get(tag);
     this.#tags.delete(tag);
     return old;
   }
 
+  /**
+   * Get the decoder for a given tag number.
+   *
+   * @param tag The tag number.
+   * @returns The decoder function, if there is one.
+   */
+  public static getDecoder(tag: TagNumber): TagDecoder | undefined {
+    return this.#tags.get(tag);
+  }
+
+  /**
+   * Get all registered decoders as object
+   */
+  public static getAllDecoders(): {
+    [key: string]: TagDecoder;
+  } {
+    return Object.fromEntries(this.#tags);
```

Comment:
I didn't realize you could do this with Map.  Nifty.  Are you sure you want to lose the type info from TagNumber?

Reply:
Object cannot have BigInt as key so we have to loose info while converting to an object but we can return a clone of the map or the map itself just like version 1 of this library

Reply:
I think a clone of the Map might be best.